tools/autograd/derivatives.yaml

```diff
@@ -786,11 +786,11 @@
 
 - name: xlogy.Tensor(Tensor self, Tensor other) -> Tensor
   self: grad * at::xlogy((self != 0), other)
-  other: grad * self / other
+  other: grad * at::where(other.isnan() | (self != 0), self / other, zeros_like(other))
```

Comment:
Can someone confirm that this does not suffer from #17455?

Reply:
I can't comment on the specifics of #17455, but the change introduced here is in part to mitigate the presence of `nan`s where they are not expected in the backwards pass. Consider the following example.

```python
import torch as th

for x in [0., 2., float('nan')]:
    for y in [0., 1.3, float('nan')]:
        x = th.as_tensor(x).requires_grad_()
        y = th.as_tensor(y).requires_grad_()
        th.xlogy(x, y).backward()
        print(f'df/dx({x:.3f}, {y:.3f}) = {x.grad:.3f}')
        print(f'df/dy({x:.3f}, {y:.3f}) = {y.grad:.3f}')
```

Running this script on `master` and this branch and comparing the outputs yields.

```diff
diff --git a/old.txt b/new.txt
index 1141aef386..0fe45e40ed 100644
--- a/old.txt
+++ b/new.txt
@@ -1,18 +1,18 @@
 df/dx(0.000, 0.000) = 0.000
-df/dy(0.000, 0.000) = nan
+df/dy(0.000, 0.000) = 0.000
 df/dx(0.000, 1.300) = 0.000
 df/dy(0.000, 1.300) = 0.000
 df/dx(0.000, nan) = nan
 df/dy(0.000, nan) = nan
 df/dx(2.000, 0.000) = -inf
 df/dy(2.000, 0.000) = inf
 df/dx(2.000, 1.300) = -inf
 df/dy(2.000, 1.300) = 1.538
 df/dx(2.000, nan) = nan
 df/dy(2.000, nan) = nan
 df/dx(nan, 0.000) = -inf
 df/dy(nan, 0.000) = nan
 df/dx(nan, 1.300) = -inf
 df/dy(nan, 1.300) = nan
 df/dx(nan, nan) = nan
 df/dy(nan, nan) = nan
```

In other words, the presence of `nan`s in gradients should only be reduced as a result of merging this PR.